Library/Homebrew/utils/analytics.rb

```diff
@@ -1,52 +1,67 @@
-
-def analytics_anonymous_prefix_and_os
+def analytics_label
   @analytics_anonymous_prefix_and_os ||= begin
-    "#{OS_VERSION}, #{HOMEBREW_PREFIX.to_s.gsub(ENV["HOME"], "~")}"
+    os = OS_VERSION
+    prefix = ", non-/usr/local" if HOMEBREW_PREFIX.to_s != "/usr/local"
+    ci = ", CI" if ENV["CI"]
+    "#{os}#{prefix}#{ci}"
   end
 end
 
-def report_analytics(type, metadata={})
+def report_analytics(type, metadata = {})
   return unless ENV["HOMEBREW_ANALYTICS"]
+  return if ENV["HOMEBREW_NO_ANALYTICS"]
 
-  metadata_args = metadata.map do |key, value|
-    ["-d", "#{key}=#{value}"] if key && value
-  end.compact.flatten
+  args = %W[
+    --max-time 3
+    --user-agent HOMEBREW_USER_AGENT_CURL
```

Comment:
This needs to be `--user-agent #{HOMEBREW_USER_AGENT_CURL}`. Otherwise the user agent will be pretty useless. :wink: